stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -259,7 +259,7 @@ class TaskGroupBase : public TaskGroupTaskStatusRecord {
   /// Any TaskGroup always IS its own TaskRecord.
   /// This allows us to easily get the group while cancellation is propagated throughout the task tree.
   TaskGroupTaskStatusRecord *getTaskRecord() {
-    return reinterpret_cast<TaskGroupTaskStatusRecord *>(this);
```

Comment:
I don't quite understand why this cast was wrong now after the refactoring, but it was wrong.